MDANSE/Src/MDANSE/Framework/AtomSelector/selector.py

```diff
@@ -13,362 +13,234 @@
-        """
-        new_settings = self.create_default_settings()
-        new_settings["all"] = False
-
-        added = set([])
-        for k, v in self.settings.items():
-            if k == "index":
-                continue
-
-            if isinstance(v, dict):
-                args = [{self._kwarg_keys[k]: i} for i in v.keys()]
-                switches = v.values()
-            else:
-                args = [{}]
-                switches = [v]
-
-            for arg, switch in zip(args, switches):
-                if not switch:
-                    continue
-
-                selection = self._funcs[k](self.trajectory, **arg)
-                if not idxs.issuperset(selection):
-                    continue
-
-                added.update(selection)
-                if isinstance(v, dict):
-                    new_settings[k][arg[self._kwarg_keys[k]]] = True
-                else:
-                    new_settings[k] = True
-
-        for idx in idxs - added:
-            new_settings["index"][idx] = True
-
-        self.settings = new_settings
-
-    def settings_to_json(self) -> str:
-        """Return the minimal json string required to achieve the same
-        settings with the settings_from_json method.
+        json_string : str
+            new selection operation in a JSON string
+        trajectory : Trajectory
+            a trajectory instance for which current_selection is defined
+        current_selection : Set[int]
+            set of currently selected atom indices
 
         Returns
         -------
-        str
-            A JSON string.
+        bool
+            True if the operation changes selection, False otherwise
+
         """
-        minimal_dict = {}
-        for k0, v0 in self.settings.items():
-            if isinstance(v0, bool) and (k0 == "all" or k0 != "all" and v0):
-                minimal_dict[k0] = v0
-            elif isinstance(v0, dict):
-                sub_list = []
-                for k1, v1 in v0.items():
-                    if v1:
-                        sub_list.append(k1)
-                if sub_list:
-                    minimal_dict[k0] = sorted(sub_list)
-        return json.dumps(minimal_dict)
-
-    def json_to_settings(self, json_string: str) -> dict[str, Union[bool, dict]]:
-        """Loads the json string and converts to a settings.
+        function_parameters = json.loads(json_string)
+        if not self.operations:
+            return True
+        operation_type = function_parameters.get("operation_type", "union")
+        selection = self.apply_single_selection(
+            function_parameters, trajectory, current_selection
+        )
+        return bool(
+            (
+                len(selection.difference(current_selection)) > 0
+                and operation_type == "union"
+            )
+            or (
+                len(current_selection.difference(selection)) > 0
+                and operation_type != "union"
+            )
+        )
```

Comment:
Give `if len(Collection) > 0` is equivalent to `if Collection` we can simplify this to:
```suggestion
        return ((selection - current_selection) and operation_type == "union") or
                (current_selection - selection) and operation_type != "union"))
```